sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -121,12 +127,39 @@ public static Resource create(Attributes attributes, @Nullable String schemaUrl)
   @Nullable
   public abstract String getSchemaUrl();
 
+  /**
+   * Returns a map of attributes that describe the resource, not associated with entites.
+   *
+   * @return a map of attributes.
+   */
+  abstract Attributes getRawAttributes();
+
+  /**
+   * Returns a collectoion of associated entities.
+   *
+   * @return a collection of entities.
+   */
+  public abstract Collection<Entity> getEntities();
+
   /**
    * Returns a map of attributes that describe the resource.
    *
    * @return a map of attributes.
    */
-  public abstract Attributes getAttributes();
+  // TODO - making this final breaks binary compatibility checks.
+  public Attributes getAttributes() {
+    // TODO - cache this.
```

Comment:
This sounds wrong because we try to represent an underlying reality with changing descriptive attributes, right? Therefore, I would expect that entity pointers being constant but the resulting attribute set being subject to change.